core/server/common/src/main/java/alluxio/master/journal/AbstractJournalSystem.java

```diff
@@ -42,6 +56,60 @@ public synchronized void stop() throws InterruptedException, IOException {
     stopInternal();
   }
 
+  @Override
+  public void addJournalSink(Master master, JournalSink journalSink) {
+    mSinkLock.writeLock().lock();
```

Comment:
would the `LockResource` pattern work here?

Reply:
Yep, updated.